NasdaqDataLink.cs

```diff
@@ -0,0 +1,194 @@
+            data.Symbol = config.Symbol;
+            var csv = line.Split(',');
+
+            if (!_isInitialized)
+            {
+                _isInitialized = true;
+                foreach (var propertyName in csv)
+                {
+                    var property = propertyName.Trim().ToLowerInvariant();
+                    data.SetProperty(property, 0m);
+                    _propertyNames.Add(property);
+                }
+                // Returns null at this point where we are only reading the properties names
+                return null;
+            }
+
+            data.Time = DateTime.ParseExact(csv[0], "yyyy-MM-dd", CultureInfo.InvariantCulture);
+            data.EndTime = data.Time + TimeSpan.FromDays(1);
+
+            for (var i = 1; i < csv.Length; i++)
+            {
+                var value = csv[i].ToDecimal();
+                data.SetProperty(_propertyNames[i], value);
+            }
+            
+            // Check if the data is price data
+            var priceKey = new List<string> {"close", "price", "settle", "value"};
+            var commonList = _propertyNames.Intersect(priceKey).ToList();
+            
+            if (commonList.Any())
+            {
+                // If so, set .Value as the first common element with price keyword
+                data.Value = (decimal)data.GetProperty(commonList[0]);
+            }
+
+            return data;
+        }
+
+        /// <summary>
+        /// Set the auth code for the Nasdaq Data Link set to the QuantConnect auth code.
+        /// </summary>
+        /// <param name="authCode"></param>
+        public static void SetAuthCode(string authCode)
+        {
+            if (string.IsNullOrWhiteSpace(authCode)) return;
+
+            _authCode = authCode;
+            IsAuthCodeSet = true;
+        }
+
+        /// <summary>
+        /// Indicates whether the data source is tied to an underlying symbol and requires that corporate events be applied to it as well, such as renames and delistings
+        /// </summary>
+        /// <returns>false</returns>
+        public override bool RequiresMapping()
+        {
+            return false;
+        }
+
+        /// <summary>
+        /// Indicates whether the data is sparse.
+        /// If true, we disable logging for missing files
+        /// </summary>
+        /// <returns>true</returns>
+        public override bool IsSparseData()
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// Gets the default resolution for this data and security type
+        /// </summary>
+        public override Resolution DefaultResolution()
+        {
+            return Resolution.Daily;
+        }
+
+        /// <summary>
+        /// Gets the supported resolution for this data and security type
+        /// </summary>
+        public override List<Resolution> SupportedResolutions()
+        {
+            return DailyResolution;
```

Comment:
I wonder if it doesn't support other resolutions, thus we should use `AllResolutions` here.